### PR TITLE
Homework #13

### PR DESCRIPTION
**Pull request by AndryNosach**

### CODE REVIEW

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/dao/SubjectDAO.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import java.util.List;
+
+@Repository
+@Transactional
+public interface SubjectDAO extends CrudRepository<Subject, Integer>{
+
+    public Subject findBySubjectName(String subject);
+
+    public Subject findSubjectById(Integer id);
+//    public int addSubject(Subject subject);
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/dao/SubjectDAO.java:18]: Commented code can be deleted

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/filters/LoggerFilter.java [hunk 2]
```diff
@@ -22,6 +22,7 @@ public void init(FilterConfig filterConfig) throws ServletException {
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
 
         HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse res = (HttpServletResponse) servletResponse;
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/filters/LoggerFilter.java:25]: You are using response?

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/service/QuizService.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import com.nosach.quizproject.entity.Quiz;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@Transactional
```
Comment (dinamio) [inline Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/service/QuizService.java:13]: Better to mark transactional methods as transactional, not whole class

**Review by dinamio — APPROVED** [review]
Looks good

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/filters/LoggerFilter.java (hunk 2 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/filters/LoggerFilter.java:25]: No, I had an idea to log response headers, but then change my mind and forot to delete. Will do.

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/dao/SubjectDAO.java (hunk 1 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/dao/SubjectDAO.java:18]: Ok, just forgot to delete.

File: Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/service/QuizService.java (hunk 3 above)
Comment (AndryNosach) [reply Nosach/web-quiz-project/src/main/java/com/nosach/quizproject/service/QuizService.java:13]: Ok

**Pull request merged**